src/Tools/dotnet-stack/ReportCommand.cs

```diff
@@ -0,0 +1,210 @@
+
+                    stackSource.ForEach((sample) =>
+                    {
+                        var stackIndex = sample.StackIndex;
+                        while (!stackSource.GetFrameName(stackSource.GetFrameIndex(stackIndex), false).StartsWith("Thread ("))
+                            stackIndex = stackSource.GetCallerIndex(stackIndex);
+
+                        // long form for: int.Parse(threadFrame["Thread (".Length..^1)])
+                        // Thread id is in the frame name as "Thread (<ID>)"
+                        string template = "Thread (";
+                        string threadFrame = stackSource.GetFrameName(stackSource.GetFrameIndex(stackIndex), false);
+                        int threadId = int.Parse(threadFrame.Substring(template.Length, threadFrame.Length - (template.Length + 1)));
+                        if (samplesForThread.TryGetValue(threadId, out var samples))
+                        {
+                            samples.Add(sample);
+                        }
+                        else
+                        {
+                            samplesForThread[threadId] = new List<StackSourceSample>() { sample };
+                        }
+                    });
+
+                    foreach (var (threadId, samples) in samplesForThread)
+                    {
+#if DEBUG
+                        Console.WriteLine($"Found {samples.Count} stacks for thread 0x{threadId:X}");
+#endif
+                        PrintStack(threadId, samples[0], stackSource);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"[ERROR] {ex.ToString()}");
+                return -1;
+            }
+            finally
+            {
+                if (File.Exists(tempNetTraceFilename))
+                    File.Delete(tempNetTraceFilename);
+                if (File.Exists(tempEtlxFilename))
+                    File.Delete(tempEtlxFilename);
+
+                if (console.GetTerminal() != null)
+                    Console.CursorVisible = true;
+            }
+
+            return 0;
+        }
+
+        private static void PrintStack(int threadId, StackSourceSample stackSourceSample, StackSource stackSource)
+        {
+            Console.WriteLine($"Thread (0x{threadId:X}):");
+            var stackIndex = stackSourceSample.StackIndex;
+            while (!stackSource.GetFrameName(stackSource.GetFrameIndex(stackIndex), verboseName: false).StartsWith("Thread ("))
+            {
+                Console.WriteLine($"  {stackSource.GetFrameName(stackSource.GetFrameIndex(stackIndex), verboseName: false)}"
+                    .Replace("UNMANAGED_CODE_TIME", "[Native Frames]")
+                    .Replace("int32", "int")
+                    .Replace("unsigned int32", "uint")
+                    .Replace("int64", "long")
+                    .Replace("unsigned int64", "ulong"));
```

Comment:
I'm not a fan of this, but I haven't found a way to manipulate the representation of the symbols from TraceEvent.  I think this is fine for the `UNMANAGED_CODE_TIME --> [Native Frames]` conversion, but it feels iffy to make list of all the replacements.  It also obviously won't handle more complex things like ``List`1[string] --> List<string>`` etc.  I'm inclined to only do the `Replace` for `UNMANAGED_CODE_TIME`.

Reply:
I wrote the comment assuming you had control over formatting the string. I agree, don't spend much effort hacking on top of strings that TraceEvent already produced. I think the right way to address this will be in the TraceEvent library, not after the fact.